addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoDeviceHandler.java

```diff
@@ -79,19 +79,24 @@ public void initialize() {
     }
 
     private void scheduleRefreshJob() {
-        logger.debug("Scheduling update channel thread in {} s", refreshStrategy.nextRunDelayInS());
-        refreshJob = scheduler.schedule(() -> {
+        long delay = refreshStrategy.nextRunDelayInS();
+        logger.debug("Scheduling update channel thread in {} s", delay);
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
             updateChannels();
-            refreshJob.cancel(false);
-            refreshJob = null;
+            if (refreshJob != null && !refreshJob.isCancelled()) {
+                logger.debug("cancel refresh job");
+                refreshJob.cancel(false);
+                refreshJob = null;
+            }
             scheduleRefreshJob();
-        }, refreshStrategy.nextRunDelayInS(), TimeUnit.SECONDS);
+        }, delay, delay, TimeUnit.SECONDS);
```

Comment:
This introduces a delay before first refresh right ?

Reply:
That's true.

Reply:
So update of the modules can be quite long upon start. It may be disappointing

Reply:
I'm pretty sure we can revert my switch from `schedule` to `scheduleWithFixedDelay` if we can handle the error from retrofit.